lld/test/wasm/lto/stub-library.s

```diff
@@ -16,6 +16,23 @@
 # RUN: wasm-ld %t.o %t/libfoo.a %p/Inputs/stub.so -o %t2.wasm
 # RUN: obj2yaml %t2.wasm | FileCheck %s
 
+## The function `foo` is declared in stub2.so and depends on `baz`, and both
+## `foo` and `baz` are defined in an LTO object. When `foo` and `baz` are
+## DCE'd and become undefined in the LTO process, wasm-ld should not try to
+## export the (nonexistent) `baz`.
+
+# RUN: llvm-mc -filetype=obj -triple=wasm32-unknown-unknown -o %t.o %s
+# RUN: mkdir -p %t
+# RUN: llvm-as %S/Inputs/foobaz.ll -o %t/foobaz.o
+# RUN: wasm-ld %t.o %t/foobaz.o %p/Inputs/stub2.so -o %t.wasm
+# RUN: obj2yaml %t.wasm | FileCheck %s --check-prefix=UNUSED
+
+## Run the same test but with foobaz.o inside of an archive file.
+# RUN: rm -f %t/libfoobaz.a
+# RUN: llvm-ar rcs %t/libfoobaz.a %t/foobaz.o
+# RUN: wasm-ld %t.o %t/libfoobaz.a %p/Inputs/stub2.so -o %t2.wasm
+# RUN: obj2yaml %t2.wasm | FileCheck %s --check-prefix=UNUSED
```

Comment:
I would hope that we could avoid duplicating all these lines and simply modify the existing lines above to use `foobaz.ll` instead of `foo.ll`?

Reply:
Then it can't be `foo` because `foo` is already being used in `stub.so` and I need a symbol that's not used anywhere. Anyway, I will remove `foo.ll` and create another file that contains all functions we need then.

Reply:
Then it becomes basically the same as the status before creating `foobaz.ll`, with only the name of `foo.ll` changed to something else though. (Also I don't think we need two different stub files then either...)

Reply:
Reverted the last two commits and changed `foo.ll` to `funcs.ll`: https://github.com/llvm/llvm-project/pull/173422/commits/a44a6c05082888cc2a60c1a572dc96c860bf1d37